pg_shard.c

```diff
@@ -641,14 +679,13 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
  * query. This new query can then be pushed down to the worker nodes.
  */
 static Query *
-RowAndColumnFilterQuery(Query *query)
+RowAndColumnFilterQuery(Query *query, List *remoteRestrictList, List *localRestrictList)
 {
 	Query *filterQuery = NULL;
 	List *rangeTableList = NIL;
-	List *whereClauseList = NIL;
-	List *whereClauseColumnList = NIL;
-	List *projectColumnList = NIL;
 	List *columnList = NIL;
+	List *localColumnList = NIL;
```

Comment:
I think `localQuery`, `localRestrictList` and `remoteRestrictList` all make sense. I think the attribute of `local` to a column isn't immediately descriptive to me. I guess the long form would be localRestrictClauseColumnList? 

In the context of this function, I was thinking the two sources of columns to be fetched: projectColumns and whereClauseColumns. This duality to me makes it a bit clearer. The combination of these two lists from the toBeFetchedColumnList.

What do you think?